### PR TITLE
testscript: ensure that temp dir isn't a symlink

### PR DESCRIPTION
**Pull request by leitzler**
I ran into issues in `govim` with the testscript matcher since it uses real paths in the logs. When running macOS `ioutil.TempDir()` returns a sym-linked directory by default.

This PR ensures that the path used is a real directory by evaluating the root test temp dir that acts as base for `$WORK`.

See full description in https://github.com/myitcv/govim/issues/308

### CODE REVIEW

**Comment by myitcv** [comment]
This LGTM but we'll wait for @rogpeppe to opine. We can't test this adequately because the macOS Travis environment is not recent enough.

**Comment by mvdan** [comment]
> When running macOS ioutil.TempDir() returns a sym-linked directory by default.

If that's the case, add a test to prove what this is fixing? The project is already tested on Mac via Travis.

**Comment by myitcv** [comment]
> If that's the case, add a test to prove what this is fixing? The project is already tested on Mac via Travis.

I think (but @leitzler will be able to confirm) this is only surfaced on Mojave.

**Comment by leitzler** [comment]
I've only tested on Mojave (10.14.6) but I don't think the current tests would catch it anyway since it's a bit of an edge case.

go-internal as well as the original go source always handles the symlink and therefor won't be affected, but as `govim` is acting from within `vim` as context where it resolves the actual path it becomes a problem.

This would end up with different paths on macOS:
```go
package main

import (
	"fmt"
	"io/ioutil"
	"os"
)

func main() {
	tmpd, err := ioutil.TempDir("", "foo")
	if err != nil {
		panic(err)
	}
	fmt.Println(tmpd)

	os.Chdir(tmpd)
	cwd, err := os.Getwd()
	if err != nil {
		panic(err)
	}
	fmt.Println(cwd)
}
```

e.g.:
```
/var/folders/xm/ls208xd174v95pgd20rn_qbh0000gn/T/foo872692457
/private/var/folders/xm/ls208xd174v95pgd20rn_qbh0000gn/T/foo872692457
```

I'll see if I can create a simple testscript that details it further.

**Comment by leitzler** [comment]
Here is an example repo: https://github.com/leitzler/testscript-symlink

It's a bit hard to create a fair example since `govim` has a custom testdriver with custom commands to compare logs, but I hope this makes more sense.

File: testscript/testscript.go [hunk 1]
```diff
@@ -150,6 +150,10 @@ func RunT(t T, p Params) {
 	if err != nil {
 		t.Fatal(err)
 	}
+	testTempDir, err = filepath.EvalSymlinks(testTempDir)
```
Comment (rogpeppe) [inline testscript/testscript.go]: Please add a comment explaining why this is needed.

Comment (leitzler) [reply testscript/testscript.go]: Struggled a bit with how to explain, but now there is a comment there at least, feel free to suggest improvements :)

**Comment by myitcv** [comment]
Comment looks good to me @leitzler - I think we should include the test, no? 

https://github.com/leitzler/testscript-symlink/blob/master/scripts/foo.txt

**Comment by leitzler** [comment]
Good idea, I added it.

**Comment by myitcv** [comment]
Thanks @leitzler 

Please can you squash the commits down and provide a descriptive commit message? 

As you can see from https://github.com/rogpeppe/go-internal/commits/master we do not use merge commits, instead relying on the commit message in the PR. When there are multiple commits in a PR (like now) GitHub simply does a concatenation of the messages... which leaves us with something less useful :)
